torch/distributed/fsdp/_optim_utils.py

```diff
@@ -990,7 +991,20 @@ def _get_param_id_to_param_from_optim_input(
     return dict(enumerate(param_id_to_param))
 
 
-def _get_flat_param_to_fqn(model: torch.nn.Module) -> Dict[nn.Parameter, str]:
+def _get_flat_param_to_fqn(model: torch.nn.Module) -> Dict[FlatParameter, str]:
+    """
+    Constructs a mapping from ``FlatParameter`` to a \"non-canonical\" fully
+    qualified name (FQN) as opposed to the FQNs returned for ``FlatParameter`` s
+    in ``_common_utils._get_param_to_fqns(...)``. We refer to these FQNs as
+    non-canonical because these parameters are not from the original module
+    but are rather swapped in by some parallelism construct, like FSDP.
```

Comment:
I do not follow what "swapped in by some parallelism construct, like FSDP" means.

The distinction seems to be that FQNs here have been cleaned via `clean_tensor_name()` to remove the FSDP prefix. Is this what you mean by "swap"?

Reply:
My understanding is that when you do use_orig_params=False, FSDP swaps out original module parameters with a FlatParameter that represents the original parameters.

This new FlatParameter will then have its own name that FSDP gave, and the module will use that instead of the original FQNs of the original params that got swapped out.

Maybe my understanding's not right here though--would be happy to rephrase